cmd/rotate_certs.go

```diff
@@ -131,6 +116,29 @@ func (rcc *rotateCertsCmd) run(cmd *cobra.Command, args []string) error {
 		return errors.Wrap(err, "parsing the api model")
 	}
 
+	if rcc.containerService.Properties.IsAzureStackCloud() {
+		writeCustomCloudProfile(rcc.containerService)
+		err = rcc.containerService.Properties.SetAzureStackCloudSpec(api.AzureStackCloudSpecParams{IsUpgrade: false, IsScale: false})
+		if err != nil {
+			return errors.Wrap(err, "error parsing the api model")
+		}
+	}
+
+	if err = rcc.getAuthArgs().validateAuthArgs(); err != nil {
```

Comment:
Why validate after writing the custom cloud profile?

Reply:
Here we are following the deploy logic